java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/Instance.java

```diff
@@ -546,145 +544,13 @@ public void setThreadHooks(Runnable threadStart, Runnable threadStop) {
         _initData.threadStop = threadStop;
     }
 
-    public Class<?> findClass(String className) {
+    Class<?> findClass(String className) {
         return Util.findClass(className, _initData.classLoader);
     }
 
-    public ClassLoader getClassLoader() {
-        return _initData.classLoader;
-    }
-
-    //
-    // For the "class resolver".
-    //
-    @Override
-    public Class<?> apply(String typeId) {
-        Class<?> c = null;
-
-        //
-        // To convert a Slice type ID into a Java class, the following steps are taken:
-        //
-        // 1. Convert the Slice type ID into a classname (e.g., ::M::X becomes M.X).
-        // 2. Check if the application has defined any package prefixes for the top-level module
-        // (e.g.: "M").    Attempt to resolve the <package-prefix>+<class-name> in order trying each
-        // defined package    prefix.
-        // 3. If the above step fails, it checks the value of Default.Package property. If found,
-        //    prepend its value to the classname. Otherwise, attempt to use the    classname
-        // directly.
-        //
-        String fullyQualifiedClassName;
-
-        //
-        // See if we've already translated this type ID before.
-        //
-        synchronized (this) {
-            fullyQualifiedClassName = _sliceTypeIdToClassMap.get(typeId);
-        }
-
-        //
-        // The fully qualified class name was already resolved, keep using it.
-        //
-        if (fullyQualifiedClassName != null) {
-            return getConcreteClass(fullyQualifiedClassName);
-        }
-
-        //
-        // It's a new type ID, so first convert it into a Java class name.
-        //
-        fullyQualifiedClassName = Util.typeIdToClass(typeId);
-
-        //
-        // See if the application defined any package prefixes with Ice.Package.MODULE property.
-        //
-        int pos = typeId.indexOf(':', 2);
-        if (pos != -1) {
-            String topLevelModule = typeId.substring(2, pos);
-            String[] packagePrefixes = _builtInModulePackagePrefixes.get(topLevelModule);
-            if (packagePrefixes == null) {
-                packagePrefixes =
-                    _initData.properties.getIcePropertyAsList("Ice.Package." + topLevelModule);
```

Comment:
Even though we document a single value, the mapping for a module was actually a list of package-prefixes. Do we want to keep this behavior?

It's mostly useful for IceMX. See #3873.

Reply:
See #3873.